ln-resource-mgr/src/incoming_channel.rs

```diff
@@ -437,4 +467,23 @@ mod tests {
 
         assert!(!bucket.add_htlc(scid, htlc_too_big).unwrap());
     }
+
+    #[test]
+    fn test_duplicate_remove_bug_addressed() {
```

Comment:
_very_ personal preference: when fixing a bug, I like to add a test demonstrating that the bug is hit and then fix the bug + update the test in the next commit.

It creates a nice git history which clearly demonstrates how the bug was being hit, and makes the fixing of the bug a bit more test-driven. This isn't very common practice from what I've seen around, so def not blocking!